ethereum/stratum/src/main/java/org/hyperledger/besu/ethereum/stratum/GetWorkProtocol.java

```diff
@@ -91,59 +77,47 @@ public boolean maybeHandle(final String initialMessage, final StratumConnection
   public void onClose(final StratumConnection conn) {}
 
   @Override
-  public void handle(final StratumConnection conn, final String message) {
-    JsonNode jsonrpcMessage = readMessage(message);
+  public void handle(
+      final StratumConnection conn, final Buffer message, final Consumer<String> sender) {
+    JsonObject jsonrpcMessage = message.toJsonObject();
     if (jsonrpcMessage == null) {
       LOG.warn("Invalid message {}", message);
       conn.close();
       return;
     }
-    JsonNode methodNode = jsonrpcMessage.get("method");
-    JsonNode idNode = jsonrpcMessage.get("id");
-    if (methodNode == null || idNode == null) {
+    String method = jsonrpcMessage.getString("method");
+    Integer id = jsonrpcMessage.getInteger("id");
+    if (method == null || id == null) {
       LOG.warn("Invalid message {}", message);
       conn.close();
       return;
     }
-    String method = methodNode.textValue();
     if ("eth_getWork".equals(method)) {
-      JsonRpcSuccessResponse response =
-          new JsonRpcSuccessResponse(idNode.intValue(), getWorkResult);
+      JsonRpcSuccessResponse response = new JsonRpcSuccessResponse(id, getWorkResult);
       try {
         String responseMessage = mapper.writeValueAsString(response);
-        conn.send(
-            "HTTP/1.1 200 OK\r\nConnection: Keep-Alive\r\nKeep-Alive: timeout=5, max=1000\r\nContent-Length: "
-                + responseMessage.length()
-                + CRLF
-                + CRLF
-                + responseMessage);
+        sender.accept(responseMessage);
       } catch (JsonProcessingException e) {
         LOG.warn("Error sending work", e);
         conn.close();
       }
     } else if ("eth_submitWork".equals(method)) {
-      JsonNode paramsNode = jsonrpcMessage.get("params");
+      JsonArray paramsNode = jsonrpcMessage.getJsonArray("params");
```

Comment:
same again, in case if it's not an array you get a ClassCastException:
https://vertx.io/docs/apidocs/io/vertx/core/json/JsonObject.html#getJsonArray-java.lang.String-

Reply:
Handled now